diff --git a/Directory.Build.props b/Directory.Build.props
index d654c18..81aa475 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -1,12 +1,12 @@
 <Project>
   <PropertyGroup>
-    <Version>2.0.1</Version>
-    <AssemblyVersion>2.0.1</AssemblyVersion>
-    <FileVersion>2.0.1</FileVersion>
+    <Version>2.0.2</Version>
+    <AssemblyVersion>2.0.2.0</AssemblyVersion>
+    <FileVersion>2.0.2.0</FileVersion>
     <Authors>Arys</Authors>
     <Copyright>Arys</Copyright>
     <LangVersion>latest</LangVersion>
-    <Configurations>Debug;SPT-3.10 Release;SPT-3.11 Release;SPT-4.0 Release</Configurations>
+    <Configurations>Debug;SPT-3.10 Release;SPT-3.11 Release;SPT-4.0 Release;SPT-4.1 Release</Configurations>
     <Platforms>AnyCPU</Platforms>
     <SolutionName>UnityToolkit</SolutionName>
     <DistributionDir>$(MSBuildThisFileDirectory)dist\</DistributionDir>
@@ -62,4 +62,12 @@
     <SptVersion>400x</SptVersion>
     <Optimize>true</Optimize>
   </PropertyGroup>
-</Project>
\ No newline at end of file
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'SPT-4.1 Release|AnyCPU' ">
+    <OutputPath>Build\SPT-4.1\</OutputPath>
+    <SptVersion>410x</SptVersion>
+    <Optimize>true</Optimize>
+    <Deterministic>true</Deterministic>
+    <DebugType>none</DebugType>
+    <DebugSymbols>false</DebugSymbols>
+  </PropertyGroup>
+</Project>
diff --git a/UnityToolkit.sln b/UnityToolkit.sln
index 18197a7..ffdfc6d 100644
--- a/UnityToolkit.sln
+++ b/UnityToolkit.sln
@@ -13,6 +13,7 @@ Global
 		SPT-3.10 Release|Any CPU = SPT-3.10 Release|Any CPU
 		SPT-3.11 Release|Any CPU = SPT-3.11 Release|Any CPU
 		SPT-4.0 Release|Any CPU = SPT-4.0 Release|Any CPU
+		SPT-4.1 Release|Any CPU = SPT-4.1 Release|Any CPU
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
 		{C36D0C27-A025-43E5-B4F2-DFC0B17410C1}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
@@ -23,6 +24,8 @@ Global
 		{C36D0C27-A025-43E5-B4F2-DFC0B17410C1}.SPT-3.11 Release|Any CPU.Build.0 = SPT-3.11 Release|Any CPU
 		{C36D0C27-A025-43E5-B4F2-DFC0B17410C1}.SPT-4.0 Release|Any CPU.ActiveCfg = SPT-4.0 Release|Any CPU
 		{C36D0C27-A025-43E5-B4F2-DFC0B17410C1}.SPT-4.0 Release|Any CPU.Build.0 = SPT-4.0 Release|Any CPU
+		{C36D0C27-A025-43E5-B4F2-DFC0B17410C1}.SPT-4.1 Release|Any CPU.ActiveCfg = SPT-4.1 Release|Any CPU
+		{C36D0C27-A025-43E5-B4F2-DFC0B17410C1}.SPT-4.1 Release|Any CPU.Build.0 = SPT-4.1 Release|Any CPU
 		{B876D9D8-DF82-45DC-8B32-92BA269A8190}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
 		{B876D9D8-DF82-45DC-8B32-92BA269A8190}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{B876D9D8-DF82-45DC-8B32-92BA269A8190}.SPT-3.10 Release|Any CPU.ActiveCfg = SPT-3.10 Release|Any CPU
@@ -31,6 +34,8 @@ Global
 		{B876D9D8-DF82-45DC-8B32-92BA269A8190}.SPT-3.11 Release|Any CPU.Build.0 = SPT-3.11 Release|Any CPU
 		{B876D9D8-DF82-45DC-8B32-92BA269A8190}.SPT-4.0 Release|Any CPU.ActiveCfg = SPT-4.0 Release|Any CPU
 		{B876D9D8-DF82-45DC-8B32-92BA269A8190}.SPT-4.0 Release|Any CPU.Build.0 = SPT-4.0 Release|Any CPU
+		{B876D9D8-DF82-45DC-8B32-92BA269A8190}.SPT-4.1 Release|Any CPU.ActiveCfg = SPT-4.1 Release|Any CPU
+		{B876D9D8-DF82-45DC-8B32-92BA269A8190}.SPT-4.1 Release|Any CPU.Build.0 = SPT-4.1 Release|Any CPU
 	EndGlobalSection
 	GlobalSection(SolutionProperties) = preSolution
 		HideSolutionNode = FALSE
diff --git a/project/UnityToolkit.Prepatcher/UnityToolkit.Prepatcher.csproj b/project/UnityToolkit.Prepatcher/UnityToolkit.Prepatcher.csproj
index 928fc9f..11da47f 100644
--- a/project/UnityToolkit.Prepatcher/UnityToolkit.Prepatcher.csproj
+++ b/project/UnityToolkit.Prepatcher/UnityToolkit.Prepatcher.csproj
@@ -5,50 +5,49 @@
     <ProjectGuid>{B876D9D8-DF82-45DC-8B32-92BA269A8190}</ProjectGuid>
     <OutputType>Library</OutputType>
     <AppDesignerFolder>Properties</AppDesignerFolder>
-    <TargetFrameworkVersion>v3.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.8</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <RootNamespace>UnityToolkit.Prepatcher</RootNamespace>
     <AssemblyName>UnityToolkit-Prepatcher</AssemblyName>
   </PropertyGroup>
   <ItemGroup>
     <Compile Include="Patcher.cs" />
-<!--    <Compile Include="Properties\AssemblyInfo.cs"/>-->
+    <Compile Include="Properties\AssemblyInfo.cs"/>
   </ItemGroup>
   <ItemGroup>
-    <Reference Include="0Harmony, Version=2.7.0.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\HarmonyX.2.7.0\lib\net35\0Harmony.dll</HintPath>
+    <Reference Include="0Harmony">
+      <HintPath>$(SptSharedAssembliesDir)$(SptVersion)\0Harmony.dll</HintPath>
     </Reference>
-    <Reference Include="BepInEx, Version=5.4.20.0, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\BepInEx.BaseLib.5.4.20\lib\net35\BepInEx.dll</HintPath>
+    <Reference Include="BepInEx">
+      <HintPath>$(SptSharedAssembliesDir)$(SptVersion)\BepInEx.dll</HintPath>
     </Reference>
-    <Reference Include="Mono.Cecil, Version=0.10.4.0, Culture=neutral, PublicKeyToken=50cebf1cceb9d05e, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Mono.Cecil.0.10.4\lib\net35\Mono.Cecil.dll</HintPath>
+    <Reference Include="Mono.Cecil">
+      <HintPath>$(SptSharedAssembliesDir)$(SptVersion)\Mono.Cecil.dll</HintPath>
     </Reference>
-    <Reference Include="Mono.Cecil.Mdb, Version=0.10.4.0, Culture=neutral, PublicKeyToken=50cebf1cceb9d05e, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Mono.Cecil.0.10.4\lib\net35\Mono.Cecil.Mdb.dll</HintPath>
+    <Reference Include="Mono.Cecil.Mdb">
+      <HintPath>$(SptSharedAssembliesDir)$(SptVersion)\Mono.Cecil.Mdb.dll</HintPath>
     </Reference>
-    <Reference Include="Mono.Cecil.Pdb, Version=0.10.4.0, Culture=neutral, PublicKeyToken=50cebf1cceb9d05e, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Mono.Cecil.0.10.4\lib\net35\Mono.Cecil.Pdb.dll</HintPath>
+    <Reference Include="Mono.Cecil.Pdb">
+      <HintPath>$(SptSharedAssembliesDir)$(SptVersion)\Mono.Cecil.Pdb.dll</HintPath>
     </Reference>
-    <Reference Include="Mono.Cecil.Rocks, Version=0.10.4.0, Culture=neutral, PublicKeyToken=50cebf1cceb9d05e, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Mono.Cecil.0.10.4\lib\net35\Mono.Cecil.Rocks.dll</HintPath>
+    <Reference Include="Mono.Cecil.Rocks">
+      <HintPath>$(SptSharedAssembliesDir)$(SptVersion)\Mono.Cecil.Rocks.dll</HintPath>
     </Reference>
-    <Reference Include="MonoMod.RuntimeDetour, Version=21.12.13.1, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\MonoMod.RuntimeDetour.21.12.13.1\lib\net35\MonoMod.RuntimeDetour.dll</HintPath>
+    <Reference Include="MonoMod.RuntimeDetour">
+      <HintPath>$(SptSharedAssembliesDir)$(SptVersion)\MonoMod.RuntimeDetour.dll</HintPath>
     </Reference>
-    <Reference Include="MonoMod.Utils, Version=21.12.13.1, Culture=neutral, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\MonoMod.Utils.21.12.13.1\lib\net35\MonoMod.Utils.dll</HintPath>
+    <Reference Include="MonoMod.Utils">
+      <HintPath>$(SptSharedAssembliesDir)$(SptVersion)\MonoMod.Utils.dll</HintPath>
     </Reference>
     <Reference Include="System" />
   </ItemGroup>
   <ItemGroup>
     <None Include="App.config" />
-    <None Include="packages.config" />
   </ItemGroup>
   <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props"
           Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')"/>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets"/>
-  <Target Name="PostBuild" AfterTargets="Build;Rebuild">
+  <Target Name="PostBuild" AfterTargets="Build;Rebuild" Condition=" '$(SkipDeploy)' != 'true' ">
     <Exec Command="if not exist &quot;$(SptBepInExPatchersDir)UnityToolkit&quot; (
     mkdir &quot;$(SptBepInExPatchersDir)UnityToolkit&quot;
     )
@@ -56,17 +55,10 @@
     copy &quot;$(TargetPath)&quot; &quot;$(SptBepInExPatchersDir)UnityToolkit\$(TargetName).dll&quot;
     xcopy /y /s &quot;$(CopyToOutputDir)&quot; &quot;$(SptBepInExPatchersDir)&quot;" />
   </Target>
-  <Target Name="CreateReleaseZip" AfterTargets="PostBuild">
+  <Target Name="CreateReleaseZip" AfterTargets="PostBuild" Condition=" '$(SkipDeploy)' != 'true' ">
     <Exec Command="if not &quot;$(ConfigurationName)&quot;==&quot;Debug&quot; (
     cd /D &quot;$(SptBepInExPatchersDir)..\..\&quot;
     &quot;C:\Program Files\7-Zip\7z.exe&quot; a &quot;$(ReleaseArchive)&quot; &quot;BepInEx\patchers\UnityToolkit\*&quot;
     )" />
   </Target>
-  <Import Project="..\..\packages\BepInEx.Core.5.4.21\build\BepInEx.Core.targets" Condition="Exists('..\..\packages\BepInEx.Core.5.4.21\build\BepInEx.Core.targets')" />
-  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
-    <PropertyGroup>
-      <ErrorText>This project references NuGet package(s) that are missing on this computer. Enable NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105.The missing file is {0}.</ErrorText>
-    </PropertyGroup>
-    <Error Condition="!Exists('..\..\packages\BepInEx.Core.5.4.21\build\BepInEx.Core.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\..\packages\BepInEx.Core.5.4.21\build\BepInEx.Core.targets'))" />
-  </Target>
 </Project>
diff --git a/project/UnityToolkit.Prepatcher/Properties/AssemblyInfo.cs b/project/UnityToolkit.Prepatcher/Properties/AssemblyInfo.cs
index 8e46bfa..3c51f8d 100644
--- a/project/UnityToolkit.Prepatcher/Properties/AssemblyInfo.cs
+++ b/project/UnityToolkit.Prepatcher/Properties/AssemblyInfo.cs
@@ -7,7 +7,7 @@ using System.Runtime.InteropServices;
 [assembly: AssemblyTitle("UnityToolkit-Prepatcher")]
 [assembly: AssemblyDescription("")]
 [assembly: AssemblyConfiguration("")]
-[assembly: AssemblyCompany("")]
+[assembly: AssemblyCompany("Arys")]
 [assembly: AssemblyProduct("UnityToolkit-Prepatcher")]
 [assembly: AssemblyCopyright("Copyright © Arys 2025")]
 [assembly: AssemblyTrademark("")]
@@ -31,5 +31,6 @@ using System.Runtime.InteropServices;
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.0.0.0")]
\ No newline at end of file
+[assembly: AssemblyVersion("2.0.2.0")]
+[assembly: AssemblyFileVersion("2.0.2.0")]
+[assembly: AssemblyInformationalVersion("2.0.2")]
diff --git a/project/UnityToolkit/Patches/InjectPlayerLoopSystems.cs b/project/UnityToolkit/Patches/InjectPlayerLoopSystems.cs
index 5958cc5..e60cfff 100644
--- a/project/UnityToolkit/Patches/InjectPlayerLoopSystems.cs
+++ b/project/UnityToolkit/Patches/InjectPlayerLoopSystems.cs
@@ -23,7 +23,7 @@ public class InjectPlayerLoopSystems : ModulePatch
 	protected override MethodBase GetTargetMethod()
 	{
 		return AccessTools.Method(typeof(CustomPlayerLoopSystemsInjector),
-			nameof(CustomPlayerLoopSystemsInjector.Injection));
+			"Injection");
 	}
 
 	[PatchPostfix]
@@ -67,4 +67,4 @@ public class HelloWorldService : IStartable
 		_logger.LogInfo("Hello world! This message means that VContainer was successfully initialized.");
 	}
 }
-#endif
\ No newline at end of file
+#endif
diff --git a/project/UnityToolkit/UnityToolkit.csproj b/project/UnityToolkit/UnityToolkit.csproj
index e5f11a8..8dbd9f2 100644
--- a/project/UnityToolkit/UnityToolkit.csproj
+++ b/project/UnityToolkit/UnityToolkit.csproj
@@ -22,6 +22,9 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'SPT-4.0 Release|AnyCPU' ">
     <TargetFramework>netstandard2.1</TargetFramework>
   </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'SPT-4.1 Release|AnyCPU' ">
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
   
   <ItemGroup>
     <Compile Remove="References\**" />
@@ -116,11 +119,11 @@
     <WriteLinesToFile File="@(PluginMetadataFilePath)" Lines="@(GeneratedText)" Overwrite="true" />
   </Target>
   
-  <Target Name="PostBuild" AfterTargets="Build;Rebuild">
+  <Target Name="PostBuild" AfterTargets="Build;Rebuild" Condition=" '$(SkipDeploy)' != 'true' ">
     <Exec Command="if not exist &quot;$(SptBepInExPluginsDir)UnityToolkit&quot; (&#xA;    mkdir &quot;$(SptBepInExPluginsDir)UnityToolkit&quot;&#xA;)&#xA;&#xA;copy &quot;$(TargetPath)&quot; &quot;$(SptBepInExPluginsDir)UnityToolkit\$(TargetFileName)&quot;&#xA;copy &quot;$(ProjectDir)References\*.dll&quot; &quot;$(SptBepInExPluginsDir)UnityToolkit&quot;&#xA;xcopy /y /s &quot;$(CopyToOutputDir)&quot; &quot;$(SptBepInExPluginsDir)UnityToolkit&quot;&#xA;&#xA;if &quot;$(ConfigurationName)&quot;==&quot;Debug&quot; (&#xA;    if exist &quot;$(TargetDir)$(TargetName).pdb&quot; (&#xA;        copy &quot;$(TargetDir)$(TargetName).pdb&quot; &quot;$(SptBepInExPluginsDir)UnityToolkit\$(TargetName).pdb&quot;&#xA;    ) else (&#xA;        echo Debug symbols not found!&#xA;    )&#xA;) else (&#xA;    if exist &quot;$(SptBepInExPluginsDir)UnityToolkit\$(TargetName).pdb&quot; (&#xA;        del &quot;$(SptBepInExPluginsDir)UnityToolkit\$(TargetName).pdb&quot;&#xA;    )&#xA;)&#xA;&#xA;" />
   </Target>
   
-  <Target Name="CreateReleaseZip" AfterTargets="PostBuild">
+  <Target Name="CreateReleaseZip" AfterTargets="PostBuild" Condition=" '$(SkipDeploy)' != 'true' ">
     <Exec Command="if not &quot;$(ConfigurationName)&quot;==&quot;Debug&quot; (&#xA;    cd /D &quot;$(SptBepInExPluginsDir)..\..\&quot;&#xA;    &quot;C:\Program Files\7-Zip\7z.exe&quot; a &quot;$(ReleaseArchive)&quot; &quot;BepInEx\plugins\UnityToolkit\*&quot;&#xA;)" />
   </Target>
 </Project>
